### PR TITLE
First version of GRIB filtering and conversion notebook in Databricks

### PR DESCRIPTION
**Pull request by giltinde**
This is a simple first version of the notebook for discharge filtering by upstream area and then conversion from GRIB to Parquet. Writing to and reading from S3 is not addressed.

### CODE REVIEW

File: databricks/grib-filtering-and-conversion.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC
+# MAGIC **Open upstream area NetCDF file and restrict to area of interest**
+
+# COMMAND ----------
+
+upstream_file_path = '/dbfs/FileStore/flood/uparea_glofas_v4_0.nc'
+ds_upstream = load_dataset(upstream_file_path)
```
Comment (A-Stangeland) [inline databricks/grib-filtering-and-conversion.py:80]: Maybe unnecessairy to write a new function to open a xarray dataset. The function name is misleading.

File: databricks/grib-filtering-and-conversion.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    longitude=slice(lon_min  - buffer, lon_max  + buffer))
+
+# COMMAND ----------
+
+# MAGIC %md 
+# MAGIC **Apply upstream filtering**
+
+# COMMAND ----------
+
+upstream_threshold = UPSTREAM_THRESHOLD
```
Comment (A-Stangeland) [inline databricks/grib-filtering-and-conversion.py:92]: Why?

File: databricks/grib-filtering-and-conversion.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+# MAGIC
+# MAGIC **Save to Parquet**
+
+# COMMAND ----------
+
+filtered_parquet_filename = f'filtered_{year}_{month}_{day}_{leadtime_hour}.parquet'
+# filtered_parquet_folder = f'data/{year}-{month}-{day}/' # Should create a folder first
+filtered_parquet_folder = ''
+filtered_parquet_file_path = filtered_parquet_folder + filtered_parquet_filename
+converter.dataframe_to_parquet(filtered_df, filtered_parquet_file_path)
```
Comment (A-Stangeland) [inline databricks/grib-filtering-and-conversion.py:120]: We should change this to write to S3.

**Comment by giltinde** [comment]
The notebooks now use the mounted cloud storage. Addressing `load_dataset` still needs to be done.

File: databricks/grib-filtering-and-conversion.py (hunk 3 above)
Comment (giltinde) [reply databricks/grib-filtering-and-conversion.py:139]: Done

File: databricks/grib-filtering-and-conversion.py (hunk 2 above)
Comment (giltinde) [reply databricks/grib-filtering-and-conversion.py]: Fixed by using config file storing constants

File: databricks/grib-filtering-and-conversion.py (hunk 1 above)
Comment (giltinde) [reply databricks/grib-filtering-and-conversion.py]: Renamed `load_dataset` to `open_dataset`. Opted for explicit engine selection through custom `open_dataset` over xarray's auto-detection for better predictability and control.

**Pull request merged**